lib/std/RingBuffer.zig

```diff
@@ -11,14 +11,15 @@
 
 const Allocator = @import("std").mem.Allocator;
 const assert = @import("std").debug.assert;
+const copy = @import("std").mem.copy;
```

Comment:
Could any of the callsites be replaced with `@memcpy` instead?

`@memcpy` is better when you can assert that the source and destination have no overlap.

Reply:
This is currently an issue for the write function due to block.zig's `executeSequenceRingBuffer` which writes slices taken from the RingBuffer into the RingBuffer itself (diff on these lines in this PR) - this does seem error-prone but obviously can be more performant than alternatives avoiding this

Reading the RingBuffer into itself seems like pretty horrible behaviour, which can be achieved manually/more explicitly via slice->write as in block.zig if desired, so have changed read functions to using @memcpy